frontend/javascripts/dashboard/dataset/dataset_settings_view.tsx

```diff
@@ -80,122 +62,87 @@ export type FormData = {
-  static contextType = defaultContext;
-  declare context: React.ContextType<typeof defaultContext>;
-
-  state: State = {
-    hasUnsavedChanges: false,
-    dataset: null,
-    datasetDefaultConfiguration: null,
-    isLoading: true,
-    messages: [],
-    activeDataSourceEditMode: "simple",
-    activeTabKey: "data",
-    savedDataSourceOnServer: null,
-  };
-
-  async componentDidMount() {
-    await this.fetchData();
-    sendAnalyticsEvent("open_dataset_settings", {
-      datasetName: this.state.dataset ? this.state.dataset.name : "Not found dataset",
-    });
-
-    const beforeUnload = (
-      newLocation: HistoryLocation<unknown>,
-      action: HistoryAction,
-    ): string | false | void => {
-      // Only show the prompt if this is a proper beforeUnload event from the browser
-      // or the pathname changed
-      // This check has to be done because history.block triggers this function even if only the url hash changed
-      if (action === undefined || newLocation.pathname !== window.location.pathname) {
-        const { hasUnsavedChanges } = this.state;
-
-        if (hasUnsavedChanges) {
-          window.onbeforeunload = null; // clear the event handler otherwise it would be called twice. Once from history.block once from the beforeunload event
-
-          this.blockTimeoutId = window.setTimeout(() => {
-            // restore the event handler in case a user chose to stay on the page
-            // @ts-ignore
-            window.onbeforeunload = beforeUnload;
-          }, 500);
-          return messages["dataset.leave_with_unsaved_changes"];
-        }
-      }
-      return;
-    };
-
-    this.unblock = this.props.history.block(beforeUnload);
-    // @ts-ignore
-    window.onbeforeunload = beforeUnload;
-  }
-
-  componentWillUnmount() {
-    this.unblockHistory();
-  }
-
-  unblockHistory() {
+const DatasetSettingsView: React.FC<DatasetSettingsViewProps> = ({
+  datasetId,
+  isEditingMode,
+  onComplete,
+  onCancel,
+}) => {
+  const [form] = Form.useForm();
+  const queryClient = useQueryClient();
+  const isUserAdmin = useWkSelector((state) => state.activeUser?.isAdmin || false);
+
+  const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
+  const [dataset, setDataset] = useState<APIDataset | null | undefined>(null);
+  const [datasetDefaultConfiguration, setDatasetDefaultConfiguration] = useState<
+    DatasetConfiguration | null | undefined
+  >(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [activeDataSourceEditMode, setActiveDataSourceEditMode] = useState<"simple" | "advanced">(
+    "simple",
+  );
+  const [activeTabKey, setActiveTabKey] = useState<TabKey>("data");
+  const [savedDataSourceOnServer, setSavedDataSourceOnServer] = useState<
+    APIDataSource | null | undefined
+  >(null);
+
+  const unblockRef = useRef<UnregisterCallback | null>(null);
+  const blockTimeoutIdRef = useRef<number | null>(null);
+
+  const unblockHistory = useCallback(() => {
```

Comment:
I think, this can be deleted completely. the refs are only set to null (and never anything else) and the unblock stuff should happen by the new useBeforeUnload hook once the component is unmounted (which should happen on cancel).